src/pudl_archiver/depositors/zenodo/depositor.py

```diff
@@ -0,0 +1,609 @@
+                    logging purposes.
+                parse_json: whether or not to always parse the response as a
+                    JSON object. Default to True.
+
+            Returns:
+                Either the parsed JSON or the raw aiohttp.ClientResponse object.
+            """
+            logger.info(f"{method} {url} - {log_label}")
+
+            async def run_request():
+                # Convert all urls to str to in case they are pydantic Url types
+                response = await session._request(method, str(url), **kwargs)
+                if response.status >= 400:
+                    if response.headers["Content-Type"] == "application/json":
+                        json_resp = await response.json()
+                        raise ZenodoClientError(
+                            status=response.status,
+                            message=json_resp.get("message"),
+                            errors=json_resp.get("errors"),
+                        )
+                    message = await response.text()
+                    raise ZenodoClientError(
+                        status=response.status,
+                        message=message,
+                    )
+                if parse_json:
+                    return await response.json()
+                return response
+
+            response = await retry_async(
+                run_request,
+                retry_on=(
+                    aiohttp.ClientError,
+                    asyncio.TimeoutError,
+                    ZenodoClientError,
+                ),
+                retry_count=retry_count,
+            )
+            return response
+
+        return requester
+
+
+class ZenodoPublishedDeposition(ZenodoBaseDepositionInterface, PublishedDeposition):
+    """Interface to Published Zenodo deposition."""
+
+    @classmethod
+    async def get_latest_version(
+        cls,
+        dataset: str,
+        session: aiohttp.ClientSession,
+        run_settings: RunSettings,
+    ) -> "ZenodoPublishedDeposition":
+        """Create a new ZenodoDepositor.
+
+        Args:
+            dataset: Name of dataset to archive.
+            session: HTTP handler - we don't use it directly, it's wrapped in self._request.
+            run_settings: Settings from CLI.
+        """
+        self = cls(dataset_id=dataset, settings=run_settings)
+        self._session = session
+        self._request = self._make_requester(session)
+        self._dataset_settings_path = (
+            importlib.resources.files("pudl_archiver.package_data") / "zenodo_doi.yaml"
+        )
+
+        if not self.settings.initialize:
+            self = self.model_copy(
+                update={"deposition": await self._get_deposition(self.doi)}
+            )
+        return self
+
+    async def open_draft(self) -> "ZenodoDraftDeposition":
+        """Open a new draft deposition to make edits."""
+        return await ZenodoDraftDeposition.new_draft(self)
```

Comment:
I think it might be nice to have the "create a new draft from a published dep" logic live in the published dep, something like

```python
def open_draft(self):
    new_record_info = self.api_client.create_new_record(...)
    return ZenodoDraftDeposition(doi=..., api_client=...) # which then doesn't have to have any custom init stuff
```

Reply:
Agreed, changed!